docs/appendices/release-notes/5.3.5.rst

```diff
@@ -47,4 +47,11 @@ See the :ref:`version_5.3.0` release notes for a full list of changes in the
 Fixes
 =====
 
-None
+- Fixed ``IndexOutOfBoundsException`` caused by an ``IS [NOT] NULL`` filter on
+  a sub-column of an object or object array in a ``WHERE`` clause, e.g. ::
+
+    CREATE TALE test (o1 ARRAY(OBJECT AS (col INT)), o2 OBJECT);
+    SELECT * FROM test WHERE o1[1]['col'] IS NULL;
+    IndexOutOfBoundsException[Index: 1 Size: 1]
+    SELECT * FROM test AS T WHERE T.o2['unknown_col'] IS NOT NULL;
```

Comment:
it the alias as `T` here necessary? otherwise please simplify to:
```
SELECT * FROM test WHERE o2['unknown_col'] IS NOT NULL;
```

Additionally please use something to ident the exception lines like:
```
=> IndexOutOfBoundsException[Index: 1 Size: 1]
```

Reply:
> it the alias as `T` here necessary? otherwise please simplify to:
> 
> ```
> SELECT * FROM test WHERE o2['unknown_col'] IS NOT NULL;
> ```

Yes, the alias is required, have not actually look into why, will have a quick look.

> Additionally please use something to ident the exception lines like:
> 
> ```
> => IndexOutOfBoundsException[Index: 1 Size: 1]
> ```

Thanks, will do that.

Reply:
> Yes, the alias is required, have not actually look into why, will have a quick look.

It is due to a known bug, where the unknown object keys from aliased relations are not caught in analysis stage:
```
cr> select o2['unknown'] from test;                                                                                                               
ColumnUnknownException[Column o2['unknown'] unknown] -- as expected

cr> select o2['unknown'] from (select * from test) as T;                                                                                          
+---------------+
| o2['unknown'] | -- returns results (existing bug)
+---------------+
+---------------+
SELECT 0 rows in set (0.003 sec)
```